spring-grpc-core/src/main/java/org/springframework/grpc/client/ClientInterceptorsConfigurer.java

```diff
@@ -51,10 +54,29 @@ public ClientInterceptorsConfigurer(ApplicationContext applicationContext) {
 	 */
 	protected void configureInterceptors(ManagedChannelBuilder<?> builder, List<ClientInterceptor> interceptors,
 			boolean mergeWithGlobalInterceptors) {
+		this.configureInterceptors(builder, interceptors, mergeWithGlobalInterceptors, null);
+	}
+
+	/**
+	 * Configure a {@link ManagedChannelBuilder} with client interceptors.
+	 * @param builder the builder to configure
+	 * @param interceptors the non-null list of interceptors to be applied to the channel
+	 * @param mergeWithGlobalInterceptors whether the provided interceptors should be
+	 * blended with the global interceptors.
+	 * @param factory the channel factory used to filter global interceptors; no filtering
+	 * if {@code null}
+	 */
+	protected void configureInterceptors(ManagedChannelBuilder<?> builder, List<ClientInterceptor> interceptors,
+			boolean mergeWithGlobalInterceptors, GrpcChannelFactory factory) {
 		// Add global interceptors first
 		List<ClientInterceptor> allInterceptors = new ArrayList<>(this.globalInterceptors);
 		// Add specific interceptors
 		allInterceptors.addAll(interceptors);
+		// Filter all interceptors
+		if (factory != null) {
+			allInterceptors.removeIf(interceptor -> this.interceptorFilters.stream()
+				.anyMatch(filter -> !filter.shouldInclude(interceptor, factory)));
+		}
```

Comment:
Consider this scenario of filter beans configured in order:

```
Filter1: interceptor != Foo
Filter2: interceptor == Foo
```
Given an input interceptor of Foo, the current impl would not include Foo.

I could see a case where an app may define a default global filter bean and then users may override that with more specific filter beans (via `@Order`).